### PR TITLE
M44 Combat Revolver

### PR DESCRIPTION
**Pull request by musicmanvr**
## About the PR

Adds the M44 revolver from CM13
- requires #3078 

## Why / Balance

The revolver acts as a good third option between the rapid firing m1984 and reliability of the m77.

## Media

- [ X ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

image

## Breaking changes

**Changelog**

<!--
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:
- add: Ported the M44 combat revolver, find it in your nearest vendor!

### CODE REVIEW

**Comment by musicmanvr** [comment]
The stock doesn't work for some reason, but more importantly, should I add buyable crates for Marksman ammo? I'm not sure how to balance the distribution of marksman without it being broken.

**Comment by Siigiil** [comment]
> The stock doesn't work for some reason, but more importantly, should I add buyable crates for Marksman ammo? I'm not sure how to balance the distribution of marksman without it being broken.

Specify in what way the stock doesn't work, please. I'll fix it in the next attachments PR.

Additionally, all new prototype IDs should use RMC as the prefix instead of CM, and old IDs are slowly being moved to it.

Probably also a good idea to make a separate folder and file for revolvers. It's *technically* a different weapon category. And since revolvers don't inherit from pistols, it doesn't really make sense to have both in the same file.

**Comment by musicmanvr** [comment]
> > The stock doesn't work for some reason, but more importantly, should I add buyable crates for Marksman ammo? I'm not sure how to balance the distribution of marksman without it being broken.
> 
> Specify in what way the stock doesn't work, please. I'll fix it in the next attachments PR.
> 
> Additionally, all new prototype IDs should use RMC as the prefix instead of CM, and old IDs are slowly being moved to it.
> 
> Probably also a good idea to make a separate folder and file for revolvers. It's _technically_ a different weapon category. And since revolvers don't inherit from pistols, it doesn't really make sense to have both in the same file.

when the stock is removed i believe it still thinks that it needs to be extended for the gun to shoot, right now it just refuses to shoot with a blank error message

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml:45]:
```suggestion
    capacity: 7
```
It does in fact hold 7 bullets in the revolver

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
All new content IDs should be RMC and not CM

Also make sure the revolver spawns empty

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml
Comment (musicmanvr) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml:45]: this is the base revolver

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/SpeedLoaders/revolver.yml
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/SpeedLoaders/revolver.yml:5]: order should be
Parent
Id
Name

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/pistol.yml
Comment (QuietlyWhisper) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/pistol.yml:55]:
```suggestion
    proto: CMBulletRevolver44Marksman
    deleteOnSpawn: true
```

**Review by QuietlyWhisper — CHANGES_REQUESTED** [review]
We'll need to do a some work on revolver system because it contains a lot of unintended ss14 mechanics that I'm not sure are removable with yml but you could still try if it's possible
- empty cartridges should not spawn
- wielding should not cycle which cylinder you're on

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/pistol.yml
Comment (QuietlyWhisper) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Cartridges/pistol.yml:55]: This won't affect the cartridges spawning but is probably needed for the system change

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/pistol.yml
Comment (musicmanvr) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/pistol.yml:42]: out of curiosity, where are you getting the number from? I checked the wiki and it said 55

**Comment by musicmanvr** [comment]
This PR is waiting on PR #3078, closing until that pr is merged.

**Comment by DrSmugleaf** [comment]
https://github.com/RMC-14/RMC-14/pull/3078 is merged now, feel free to reopen whenever

**Comment by musicmanvr** [comment]
WE ARE SO BACK!!!

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/pistol.yml
Comment (musicmanvr) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Ammunition/Projectiles/pistol.yml]: added

**Comment by musicmanvr** [comment]
Will all of the changed added, I believe that the 44 magnum is ready for a second review.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml
Comment (Siigiil) [inline Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml:65]: The ROF is wrong. The M44's fire delay is `FIRE_DELAY_TIER_7`, which is 0.5 and equals a fire rate of 2.0.

**Review by Siigiil — COMMENTED** [review]
The scatter and recoil are also wrong, but that's less critical and will be partially reworked soon, so there isn't much point in changing them now.

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml
Comment (musicmanvr) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml]: fixed

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml
Comment (musicmanvr) [reply Resources/Prototypes/_RMC14/Entities/Objects/Weapons/Guns/Revolvers/marine_revolvers.yml:96]: the bipod was a reference to a very particular image of a revolver with a bipod and 8x scope, but if it doesn't fit, I can remove it

I swear i took the stabilizer out, fixing that now

from what i can tell:
need to remove the 4x, gyro, add the flashlight and that should do it

**Review by DrSmugleaf — APPROVED** [review]
Thanks for the PR!

**Pull request merged**
